### PR TITLE
ARROW-9031: [R] Implement conversion from Type::UINT64 to R vector

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

**Review by pitrou — COMMENTED** [review]
Why does `int64` use `Converter_Int64` but the other types are handled differently?

File: r/tests/testthat/test-Array.R [hunk 1]
```diff
@@ -385,6 +385,14 @@ test_that("Array<int8>$as_vector() converts to integer (ARROW-3794)", {
   expect_equal(a$as_vector(), 0:255)
 })
 
+test_that("Arrays of uint{32,64} convert to numeric", {
+  u32 <- Array$create(1L)$cast(uint32())
+  expect_identical(as.vector(u32), 1)
+
+  u64 <- Array$create(1L)$cast(uint64())
+  expect_identical(as.vector(u64), 1)
+})
```
Comment (pitrou) [inline r/tests/testthat/test-Array.R:394]: Those tests are thorough!

**Comment by nealrichardson** [comment]
> Why does `int64` use `Converter_Int64` but the other types are handled differently?

Historical reasons I suppose. See also this discussion: https://github.com/apache/arrow/pull/7441#discussion_r440846641

**Comment by nealrichardson** [comment]
Deferring to https://jira.apache.org/jira/browse/ARROW-9083 questions of whether there are better ways to convert. This at least no longer errors and gives something usable.

**Pull request closed without merging**